Automattic/wp-calypso
# Gutenboarding: Accordion-style plans grid.

## Changes proposed in this Pull Request

* Accordion-style plans grid.

**Detailed Changes:**

- We now split the rendering of PlanItem, one for primary plan, one for all other plans (it used to be in a single loop). Primary plan could either mean the default premium plan or the recommended plan based on the features user chosen from the feature picker. See https://github.com/Automattic/wp-calypso/pull/44299/files#diff-ef06ad3b1ea3e3d50b5d84d92780d2adR39-R48.
  - The `badge` param is introduced to customize what to show on the primary plan's badge. When default premium plan is shown, the badge display **Popular**. When a recommended plan is shown, the badge displays **Recommended for you**.
- All plans toggling (Show all plans/collapse all plans button) is now the responsibility of the parent component `PlansTable` instead of child component `PlanItem`, see https://github.com/Automattic/wp-calypso/pull/44299/files#diff-ef06ad3b1ea3e3d50b5d84d92780d2adR50-R63. `onToggleExpandAll` and `allPlansExpanded` param has been removed, see https://github.com/Automattic/wp-calypso/pull/44299/files#diff-ef06ad3b1ea3e3d50b5d84d92780d2adR50-R63.
- "Custom domain not included" is now always shown under Free plan's feature list.
- Added plan descriptions "Best for freelancers, best for online stores, etc.".
- PlanItem component at `plan-tables\plan-item.tsx` is now moved out to `plan-item\index.tsx` with it's own stylesheet specific to the component itself.


## Testing instructions

* Go to `/new`.
* Test plans grid in different viewport sizes, make sure they match similarly to design.
* Test toggling "Show all plans / Collapse all plans" button.
  * Manually expanding/collapsing each of the plan items should cause the toggle button to automatically switch "Show all plans" to "Collapse all plans" (and vice versa).
* Test the domain picker button on the feature list with:
  * No/free domain selected.
  * Paid domain selected.
* Ensure all "Choose" verb has been renamed to "Select".

## Screenshots
![2020-07-22_17-07-41](https://user-images.githubusercontent.com/1287077/88193543-16284d00-cc3e-11ea-934d-0400cb7a476a.png)
![2020-07-22_17-08-03](https://user-images.githubusercontent.com/1287077/88193547-17f21080-cc3e-11ea-9667-2186a90a8ba8.png)
![2020-07-22_17-08-51](https://user-images.githubusercontent.com/1287077/88193565-1a546a80-cc3e-11ea-88db-637da34683dd.png)
![2020-07-22_17-09-06](https://user-images.githubusercontent.com/1287077/88193568-1b859780-cc3e-11ea-836c-53a743e0cd45.png)


Fixes #44201
